resteasy-servlet-initializer/src/main/java/org/jboss/resteasy/plugins/servlet/ResteasyServletInitializer.java

```diff
@@ -95,51 +85,91 @@ protected void handleNoApplicationClass(Set<Class<?>> providers, Set<Class<?>> r
 
    }
 
+   private Set<ServletRegistration> getServletsForApplication(Class<?> applicationClass, ServletContext servletContext)
+   {
+      Set<ServletRegistration> set = new HashSet<>();
+      ServletRegistration reg = servletContext.getServletRegistration(applicationClass.getName());
+      if (reg != null && reg.getMappings().size() == 1)

```

Comment:
I'm just speculating here, not objecting.

In the JIRA you mentioned having two different servlets for the same Application. I suppose, in general, you would have two explicit <servlet/> declarations, but I gather this is legal:
```
<web-app>
   <servlet-mapping>
      <servlet-name>org.example.MyApplication</servlet-name>
      <url-pattern>/myresources/*</url-pattern>
      <url-pattern>/secure/myresources/*</url-pattern>
   </servlet-mapping>
</web-app>
```
The JAX-RS spec doesn't seem to say anything about that, so I guess it would be a new feature. Is it worth doing for RESTEasy 4.x?

Reply:
I've never seen that done before, but I think it is a good idea.  There might need to be more corresponding changes to account for different mapped paths though.  I can open up a new feature JIRA for allowing a single app to be mapped to multiple paths without needing multiple servlets.  Good idea!

Reply:
> The JAX-RS spec doesn't seem to say anything about that, so I guess it would be a new feature. Is it worth doing for RESTEasy 4.x?

yes, we're currently planning to use RESTEasy 4.5.X in Open Liberty.